docs/source/developer_tutorial.ipynb

```diff
@@ -0,0 +1,253 @@
+    "\n",
+    "**When developing, please follow this style guide: https://llego.dev/posts/writing-clean-pep-8-compliant-code-better-collaboration/**"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "aadabb8e",
+   "metadata": {},
+   "source": [
+    "Below a brief over view of the kinisi codebase is given to aid the onboarding of new developers."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0c331029",
+   "metadata": {},
+   "source": [
+    "# DiffusionAnalyzer and JumpDiffusionAnalyzer"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3799b256",
+   "metadata": {},
+   "source": [
+    "Within `kinisi` the main classes the end user will interact with are [`DiffusionAnalyzer`](https://github.com/kinisi-dev/kinisi/blob/scipp/kinisi/diffusion_analyzer.py#L20), [`JumpDiffusionAnalyzer`](https://github.com/kinisi-dev/kinisi/blob/scipp/kinisi/jump_diffusion_analyzer.py#L20) and [`ConductivityAnalyzer`](https://github.com/kinisi-dev/kinisi/blob/scipp/kinisi/conductivity_analyzer.py#L20)\n",
+    "\n",
+    "These are typically called with 3 main methods depending on the type of trajectory these are:\n",
+    "\n",
+    "* [`from_universe`](https://github.com/kinisi-dev/kinisi/blob/scipp/kinisi/diffusion_analyzer.py#L87) - For MDAnalysis Universe\n",
+    "* [`from_xdatcar`](https://github.com/kinisi-dev/kinisi/blob/scipp/kinisi/diffusion_analyzer.py#L35) - For Pymatgen Structures\n",
+    "* [`from_ase`](https://github.com/kinisi-dev/kinisi/blob/scipp/kinisi/diffusion_analyzer.py) - For ASE objects\n",
+    "\n",
+    "For instance, [`DiffusionAnalyzer.from_xdatcar`](https://github.com/kinisi-dev/kinisi/blob/scipp/kinisi/diffusion_analyzer.py#L35) can be called via"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "30b74559",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "import matplotlib.pyplot as plt\n",
+    "import scipp as sc\n",
+    "from pymatgen.io.vasp import Xdatcar\n",
+    "from kinisi.analyze import DiffusionAnalyzer\n",
+    "rng = np.random.RandomState(42)\n",
+    "\n",
+    "params = {'specie': 'Li',\n",
+    "          'time_step': 2.0 * sc.Unit('fs'),\n",
+    "          'step_skip': 50 * sc.Unit('dimensionless'),\n",
+    "          }\n",
+    "\n",
+    "\n",
+    "params['dimension']= 'xy'\n",
+    "\n",
+    "xd = Xdatcar('./example_XDATCAR.gz')\n",
+    "diff = DiffusionAnalyzer.from_xdatcar(xd, **params)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4331a507",
+   "metadata": {},
+   "source": [
+    "[`DiffusionAnalyzer`](https://github.com/kinisi-dev/kinisi/blob/scipp/kinisi/diffusion_analyzer.py#L20) is a subclass of [`Analyzer`](https://github.com/kinisi-dev/kinisi/blob/scipp/kinisi/analyzer.py#L20) and so upon its initialization `Analyzer` is initialized first, `DiffusionAnalyzer` inherits from the `Analyzer`.\n",
+    "\n",
+    "```python\n",
+    "\n",
+    "class DiffusionAnalyzer(Analyzer):\n",
```

Comment:
I am not convinced these parts are necessary given the text above and the hyperlinking.